src/lib/datocms/index.ts

```diff
@@ -16,12 +17,37 @@ type DatocmsRequest = {
   variables?: { [key: string]: string };
   retryCount?: number;
 };
+
+/**
+ * Expect return value of specified Query to be available, i.e. non-nullable.
+ */
+type Assert<Query> = Required<{ [key in keyof Query]: NonNullable<Query[key]> }>;
+
 /**
  * Makes a request to the DatoCMS GraphQL API using the provided query and variables.
  * It has authorization, environment and drafts (preview) pre-configured.
  * It has a retry mechanism in case of rate-limiting, based on DatoCMS API utils. @see https://github.com/datocms/js-rest-api-clients/blob/f4e820d/packages/rest-client-utils/src/request.ts#L239C13-L255
+ * 
+ * @template {Object} Query - The expected response data structure
+ * @template {boolean} AssertReturnValue - Whether to assert non-nullability of the return value, defaults to true for static output
+ * 
+ * @param options - Request options
+ * @param options.query - The GraphQL query document
+ * @param options.variables - Variables to pass to the GraphQL query
+ * @param options.retryCount - Number of retry attempts in case of rate limiting
+ * 
+ * @returns A promise resolving to the query result, with optional type assertion
  */
-export const datocmsRequest = async <T>({ query, variables = {}, retryCount = 1 }: DatocmsRequest): Promise<T> => {
+export async function datocmsRequest<
+  Query,
+  AssertReturnValue extends boolean = typeof output extends 'static' ? true : false
```

Comment:
Infer the default behavior of route rendering from the value of output (`as const` helps by getting this as a type).